test/UnitTestHelpers.hs

```diff
@@ -0,0 +1,352 @@
+  { isymbol  = 'x'
+  , iname    = "12345678901234567890123"
+  , ifreq    = [ (UNREPORTED_INVENTORY, 1) ]
+  , iflavour = zipPlain [Green]
+  , icount   = 1 + 1 `Dice.d` 2
+  , irarity  = [(1, 50), (10, 1)]
+  , iverbHit = "hit"
+  , iweight  = 300
+  , idamage  = 1 `Dice.d` 1
+  , iaspects = [ AddSkill Ability.SkHurtMelee $ -16 * 5
+                , SetFlag Ability.Fragile
+                , toVelocity 70 ]
+  , ieffects = []
+  , idesc    = "A really cool test item."
+  , ikit     = []
+  }
+
+testActorWithItem :: Actor
+testActorWithItem = 
+  testActor { beqp = EM.singleton testItemId (1,[])}
+
+-- stublike state instance that should barely function for testing
+stubState :: State
+stubState = let singletonFactionUpdate _ = EM.singleton testFactionId testFaction
+                singletonDungeonUpdate _ = EM.singleton testLevelId stubLevel
+                singletonActorDUpdate _ = EM.singleton testActorId testActor
+                singletonActorMaxSkillsUpdate _ = EM.singleton testActorId Ability.zeroSkills
+                copsUpdate oldCOps = oldCOps{corule=((corule oldCOps){rWidthMax=testLevelDimension, rHeightMax=testLevelDimension})}
+                stateWithMaxLevelDimension = updateCOpsAndCachedData copsUpdate emptyState
+                stateWithFaction = updateFactionD singletonFactionUpdate stateWithMaxLevelDimension
+                stateWithActorD = updateActorD singletonActorDUpdate stateWithFaction
+                stateWithActorMaxSkills = updateActorMaxSkills singletonActorMaxSkillsUpdate stateWithActorD
+                stateWithDungeon = updateDungeon singletonDungeonUpdate stateWithActorMaxSkills
+            in stateWithDungeon
+
+testStateWithItem :: State
+testStateWithItem = let swapToItemActor _ = EM.singleton testActorId testActorWithItem
+                     in updateActorD swapToItemActor stubState
+
+emptyCliState :: CliState
+emptyCliState = CliState
+  { cliState = emptyState 
+  , cliClient = emptyStateClient $ toEnum 0
+  , cliSession = Nothing
+  -- , cliDict = undefined 
+  -- , cliToSave = undefined 
+  }  
+
+stubSessionUI :: SessionUI
+stubSessionUI = (emptySessionUI stubUIOptions) 
+  { sactorUI = EM.singleton testActorId ActorUI { bsymbol='j', bname="Jamie", bpronoun="he/him", bcolor=BrCyan }
+  , schanF = fchanFrontendStub
+  } 
+
+stubCliState :: CliState
+stubCliState = CliState
+  { cliState = stubState
+  , cliClient = (emptyStateClient testFactionId) { soptions = stubClientOptions, sfper = EM.singleton testLevelId emptyPer }
+  , cliSession = Just (stubSessionUI {sxhair = Just (TPoint TUnknown testLevelId (Point 1 0))}) --(TVector Vector {vx=1, vy=0})}) -- (TNonEnemy (toEnum 1))})-- 
+  }
+
+testCliStateWithItem :: CliState
+testCliStateWithItem = stubCliState { cliState = testStateWithItem }
+
+-- | Client state mock transformation monad.
+newtype CliMock a = CliMock
+  { runCliMock :: StateT CliState IO a }  -- we build off io so we can compile but we don't want to use it
```

Comment:
In a separate PR we may try to get rid of the IO. I can't see any problem right now (we'd need to to define dummy `liftIO` in some monads), but there may be one.